packages/react/src/transactions/useFollow.ts

```diff
@@ -33,36 +32,42 @@ type FollowProfilesFlowRequest = {
   followerProfileId?: string;
 };
 
-function createFollowRequestFee(followModule: FeeFollowModuleSettingsFragment): FollowRequestFee {
+function createFollowRequestFee(
+  followModule: ChargeFollowPolicy,
+  contractAddress: string,
+): FollowRequestFee {
   return {
-    amount: erc20Amount({ from: followModule.amount }),
-    contractAddress: followModule.contractAddress,
+    amount: followModule.amount,
+    contractAddress,
     recipient: followModule.recipient,
   };
 }
 
 function createFollowProfilesFlowRequest(
-  profile: ProfileFieldsFragmentWithSupportedFollowModule,
+  profile: ProfileFieldsFragment,
   activeWallet: WalletData,
   activeProfile: ProfileFieldsFragment,
 ): FollowProfilesFlowRequest {
-  const followPolicyType = getFollowPolicyTypeFromProfileFieldsFragment(profile.followModule);
-
   const baseRequest: Pick<FollowProfilesFlowRequest, 'profileId' | 'followerAddress'> = {
     profileId: profile.id,
     followerAddress: activeWallet.address,
   };
 
+  const followPolicyType = profile.followPolicy?.type ?? null;
+
   switch (followPolicyType) {
     case FollowPolicyType.CHARGE:
       invariant(
         isProfileFieldsFragmentWithFeeFollowModule(profile),
-        'Profile is not with a fee follow module',
+        'Profile must have fee follow module',
       );
 
       return {
         ...baseRequest,
-        fee: createFollowRequestFee(profile.followModule),
+        fee: createFollowRequestFee(
+          profile.followPolicy as ChargeFollowPolicy,
```

Comment:
I believe the way that `followPolicyType` is computed confused TS so it's not able to narrow down the union.

Maybe worth do an eager return if `profile.followPolicy === null` and then evaluate `profile.followPolicy.type` with the switch.

Reply:
Thinking out loud... any reason to have `profile.followPolicy` nullable at all?

Reply:
great idea, we now model each case within `FollowPolicy` so no longer need to support null - this simplifies things massively